plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -409,4 +416,18 @@ private static String hadoopPath(String path)
         }
         return path;
     }
+
+    private static Optional<Domain> getPathDomain(TupleDomain<IcebergColumnHandle> effectivePredicate)
```

Comment:
That doesn't need to return Optional.
it's either Domain.none(), Domain.all(), or some specific Domain